[WIP]Server side of sales page

soto-ogre-private opened the pull request:
# What
- 商品を出品するページのサーバーサイド
- 商品情報をフォームに入力→itemsテーブルとdeliveriesテーブルに値を保存。
- `accepts_nested_attributes_for`メソッドを使用しitemsとdeliveriesを親子関係にした。
- 画像の複数アップロードには**CarrierWave**の`mount_uploader`メソッドを使用。
- controllerのテストにはRequest Specを使用

# Why
ユーザーが商品を出品するために必要な機能なため

soto-ogre-private committed:
update

soto-ogre-private committed:
update

soto-ogre-private committed:
update

soto-ogre-private committed:
update

soto-ogre-private committed:
save value in items table & delivery table

フォームに入力された値をitems tableとdelivery tableに保存。
画像の複数アップロードにCarrierWaveを使用。

soto-ogre-private committed:
Merge branch 'Server_side_of_sales_page' of https://github.com/yunakajimayu/freemarket_sample_61c into Server_side_of_sales_page

soto-ogre-private committed:
Saved value to items table & deliveries table

itemsテーブルとcategoriesテーブルに値を保存

soto-ogre-private committed:
update

soto-ogre-private committed:
edited uploader

soto-ogre-private committed:
Create 20200218093252_create_items.rb

soto-ogre-private committed:
test code

soto-ogre-private committed:
FactoryBot saved pictures column

yunakajimayu commented on app/models/delivery.rb:
```diff
@@ -0,0 +1,38 @@
+class Delivery < ApplicationRecord
+  belongs_to :item,optional: true
+
+  enum delivery_day: [
```
#コードレビューではありません。
自身の知らなかったコードについての記録です。
enumは列挙型のことで、モデルの数値カラムに対して文字列による名前定義をマップすることができる。

yunakajimayu commented on app/models/item.rb:
```diff
@@ -1,10 +1,27 @@
   has_many :likes
+  # belongs_to :seller, class_name: "User",foreign_key: "seller_id"
+  # belongs_to :buyer, class_name: "User",foreign_key: "buyer_id"
+  mount_uploaders :pictures, PicturesUploader
```
#コードレビューではありません。
自分の知らなかったコードについての記録です。
複数画像を持つためのコード。

yunakajimayu commented on app/views/items/_form.html.haml:
```diff
@@ -0,0 +1,261 @@
+-#       .actions
+-#         = f.hidden_field :size,value: 0
+-#         = f.hidden_field :status,value: 0
+-#         = f.submit 'Save'
```
1~113のコメントアウトは必要なければ削除お願いします。

yunakajimayu commented on app/models/item.rb:
```diff
@@ -1,10 +1,27 @@
+  mount_uploaders :pictures, PicturesUploader
+  serialize :pictures, JSON
+  
+  accepts_nested_attributes_for :delivery
```
#コードレビューではありません。
accepts_nested_attributes_forは一つのフォームから複数のモデルを保存することができる。

yunakajimayu reviewed (commented):
プルリクエストありがとうございます。
テストがまだだと思いますが、現時点でのコードレビューをしました。
＃コードレビューではないものは自身の備忘録としてコメントしたものです。
気にしないでください。

yunakajimayu committed:
Update deploy.rb

soto-ogre-private committed:
Rspec code

soto-ogre-private committed:
Fix js for select tag

「配送料の負担」を選んだ後に表示される「配送方法」のセレクトタグを選択しても値が入らずエラーを吐いてしまっていたので修正しました。

soto-ogre-private committed:
Merge branch 'master' into Server_side_of_sales_page

indiamela commented:
LGTMです！

soto-ogre-private committed:
Merge branch 'Server_side_of_sales_page' of https://github.com/yunakajimayu/freemarket_sample_61c into Server_side_of_sales_page

soto-ogre-private committed:
fix

soto-ogre-private committed:
Update item.js

TOMARUMARU commented on app/assets/stylesheets/items.scss:
```diff
@@ -158,9 +158,9 @@
+            // &.seller, &.buyer {
+            //   display: none;
+            // }
             div:first-child {
```
不要なコメントアウトは消しましょう

soto-ogre-private replied:
削除しました！

TOMARUMARU commented on config/deploy.rb:
```diff
@@ -31,13 +31,21 @@
     invoke 'unicorn:restart'
   end
 
+<<<<<<< HEAD
```
こちらのファイルのコンフリクトの記述消しましょう

soto-ogre-private replied:
削除しました！

TOMARUMARU commented on config/routes.rb:
```diff
@@ -6,11 +6,12 @@
+  get 'transaction',on: :collection
+  post 'purchase', on: :collection
+  get 'done', on: :collection
+  get ""=>"items#show",on: :member
```
routingはresourcesベースで書きましょう

soto-ogre-private committed:
fix

・不要なコメントアウトの削除
・コンフリクトの解消
・商品出品ページはURLを"sell"にしたいためルーティングを設定
・showアクションのルーティングを削除(resourcesを利用する)

shingoteshima commented:
LGTMです！！
